pkg/collector/parser/receiver.go

```diff
@@ -100,6 +100,11 @@ func singlePortFromConfigEndpoint(logger logr.Logger, name string, config map[in
 	case name == "tcplog" || name == "udplog":
 		endpoint = getAddressFromConfig(logger, name, listenAddressKey, config)
 
+	// in case of kubeletstats reciever, the endpoint is ignored
+	case name == "kubeletstats":
+		endpoint = nil
```

Comment:
All you have to do is `return nil` here. No use of `nil` assignment. Also, add unit tests for the same.